src/server.js

```diff
@@ -0,0 +1,35 @@
+import 'dotenv/config'
+import { WebSocketServer } from 'ws'
+import { v4 as uuid } from 'uuid'
+import ZmqConnection from './zmq.js'
+import Client from './client.js'
+import Dashcore from '@dashevo/dashcore-lib'
+
+export default class Server {
+  #clients = []
+
+  constructor () {
+    this.zmq = new ZmqConnection()
+    this.zmq.on('rawtx', (data) => {
+      this.handleRawTransactions(data)
+    })
+  }
+
+  async start () {
+    const socket = new WebSocketServer({ port: process.env.PORT })
+
+    socket.on('connection', (socket) => {
+      socket.id = uuid()
+      const client = new Client(socket)
+      this.#clients.push(client)
+      socket.send('You are successfully connected to the Dash WebSocket server!')
+    })
+
+    console.log(`WebSocket server is running on ${process.env.SERVER_URL}`)
```

Comment:
I don't think you even need `SERVER_URL` env, because it will almost always differs from your external ip. Actually, I think your bound interface ip could be available as a callback arguments in the  .open() event. Not sure though about WebSocketServer, but fastify for sure returns that.